### DISCUSSION TITLE
Is there is any alternative for swaymsg in hyprland .

### DISCUSSION THREAD

**Discussion by kpanuragh** [hyprwm/Hyprland#2295] [General]
I just started migration from sway to hyprland also  found hyprctl but I coud't found any method to convert 
`compgen -c | grep -v fzf | sort -u | fzf --layout=reverse | xargs -r swaymsg -t command exec` 
this command to hyprctl can any on help me

**Comment by vaxerski** [comment]
`| xargs -r hyprctl dispatch exec`?

**Comment by thyeun** [comment]
How to convert swaymsg below to hyprland using ```hyprctl```, anyone possible help me up?

```
"$active" -g "$(swaymsg -t get_tree | jq -r '.. | 
```

**Reply by fufexan** [reply]
What does the command do?

**Reply by thyeun** [reply]
@fufexan It use for the ```wf-recorder```, as the full command below

```
wf-recorder --audio=alsa_output.pci-0000_08_00.6.analog-stereo.monitor --file=$HOME/Videos/$filename --geometry "$(swaymsg -t get_tree | jq -r '.. | select(.pid? and .visible?) | .rect | "\(.x),\(.y) \(.width)x\(.height)"' | slurp -c "#FFFFFF" )" >/dev/null 2>&1 &
```
i want to use it on hyrpland, but stuck on ```swaymsg -t get_tree```.

**Reply by fufexan** [reply]
What about `hyprctl activewindow -j | jq -r '"\(.at[0]),\(.at[1]) \(.size[0])x\(.size[1])"'`, which yields `6,38 788x956` (in my case)?

**Reply by thyeun** [reply]
@fufexan No error, but the active window not cover by ```slurp``` with color ```#FFFFFF```

Running in terminal, it will activated the slurp, but some how on-click at waybar, the slurp wont trigger (might be also because my code having issue)

I think your ```hyprctl activewindow``` do the magic. 

thank you so much
